### PR TITLE
Chat: keep the editor query prefix on the @-mention provider selection

### PR DESCRIPTION
**Pull request by valerybugakov**
- Keep the editor query prefix on the @-mention provider selection

## Test plan

1. Open a new chat with OpenCtx enabled.
2. Type a few words and then attempt to select an OpenCtx provider.
3. In the main branch, the chat input will be cleared upon provider selection. In this branch, it should only remove the text after @.

### CODE REVIEW

File: vscode/webviews/promptEditor/plugins/atMentions/atMentions.tsx [hunk 1]
```diff
@@ -139,6 +143,7 @@ export default function MentionsPlugin({
 
     const onSelectOption = useCallback(
         (selectedOption: MentionMenuOption, nodeToReplace: TextNode | null, closeMenu: () => void) => {
+            // console.log({ selectedOption, nodeToReplace })
```
Comment (eseliger) [inline vscode/webviews/promptEditor/plugins/atMentions/atMentions.tsx]:
```suggestion
```

File: vscode/webviews/promptEditor/plugins/atMentions/atMentions.tsx [hunk 2]
```diff
@@ -240,7 +245,8 @@ export default function MentionsPlugin({
                     },
                 })
             }}
-            menuRenderFn={(anchorElementRef, { selectOptionAndCleanUp }) => {
+            menuRenderFn={(anchorElementRef, itemProps) => {
+                const { selectOptionAndCleanUp } = itemProps
```
Comment (eseliger) [inline vscode/webviews/promptEditor/plugins/atMentions/atMentions.tsx:248]: is this diff necessary?

**Review by thenamankumar — APPROVED** [review]
LGTM

**Review by eseliger — APPROVED** [review]
Trusting you here, code doesn't look wrong but running locally didn't work because providers still don't show there for me :D

**Pull request merged**